### ISSUE TITLE
Can not inherit triggers,notifications,parameterConfig correctly from mpt-v2 template.

### ISSUE THREAD

**Issue by shuai-gao-awx** [spinnaker/spinnaker#4451]
### Issue Summary:
Can not inherit triggers,notifications,parameterConfig correctly from mpt-v2 template. I followed https://www.spinnaker.io/guides/user/pipeline/pipeline-templates/ to create a pipeline template and a pipeline configuration and added parameterConfig and trigger into the template. All seems good with `spin pi plan`, but when I browse the UI,  "Automated Triggers" and "Parameters" are empty.

### Cloud Provider(s):

Kubernetes V2

### Environment:

spinnaker version: `1.14.0`

feature flags:
 - `hal config features edit --artifacts true`
 - `hal config features edit --pipeline-templates true`
 - `hal config features edit --managed-pipeline-templates-v2-ui true`

spin cli version: `1.7.2`

### Feature Area:

MPT V2 template,  inherit

### Description:

Pipeline Template Json(newSpelTemplate.json, with parameterConfig list, triggers list, and notification list): 

```json

{
  "schema" : "v2",
  "variables" : [
    {
      "type" : "int",
      "defaultValue" : 42,
      "description" : "The time a wait stage shall pauseth",
      "name" : "timeToWait"
    }
  ],
  "id" : "newSpelTemplate",
  "protect" : false,
  "metadata" : {
    "name" : "Variable Wait",
    "description" : "A demonstrative Wait Pipeline.",
    "owner" : "example@example.com",
    "scopes" : ["global"]
  },
  "pipeline": {
    "lastModifiedBy" : "anonymous",
    "updateTs" : "0",
    "parameterConfig" : [
      {
        "default": "dev",
        "description": "The environment to deploy to",
        "hasOptions": true,
        "label": "Environment",
        "name": "ENV",
        "pinned": true,
        "options": [
          {
            "value": "dev"
          },
          {
            "value": "staging"
          },
          {
            "value": "demo"
          },
          {
            "value": "prod"
          }
        ]
      }
    ],
    "limitConcurrent": true,
    "keepWaitingPipelines": false,
    "description" : "",
    "triggers" : [
      {
        "attributeConstraints": {},
        "enabled": true,
        "payloadConstraints": {},
        "source": "testhook",
        "type": "webhook"
      }
    ],
    "notifications" : [
      {
        "address": "aa@bb",
        "cc": "cc@dd",
        "level": "pipeline",
        "message": {
          "pipeline.starting": {
            "text": "aabbcc"
          }
        },
        "type": "email",
        "when": [
          "pipeline.starting"
        ]
      }
    ],
    "stages" : [
      {
        "waitTime" : "${ templateVariables.timeToWait }",
        "name": "My Wait Stage",
        "type" : "wait",
        "refId" : "wait1",
        "requisiteStageRefIds": []
      }
    ]
  }
}

```

Pipeline Configuration Json(v2pipetest.json, with "parameterConfig" and "notifications" in "inherit" list, but no "triggers" ): 

```json

{
  "schema": "v2",
  "application": "v2pipetest",
  "name": "v2pipetest",
  "template": {
    "artifactAccount": "front50ArtifactCredentials",
    "reference": "spinnaker://newSpelTemplate",
    "type": "front50/pipelineTemplate"
  },
  "variables": {
    "timeToWait": 4
  },
  "inherit": [
    "parameterConfig",
    "notifications",
  ],
  "triggers": [],
  "parameterConfig": [],
  "notifications": [],
  "description": "",
  "stages": [],
  "exclude": []
}

```

Save template, plan the config and then save the config:

```bash

$ spin pt save -f newSpelTemplate.json
Caching oauth2 token.
Pipeline template save succeeded

$ spin pt plan -f v2pipetest.json
<output json>

$ spin pi save -f v2pipetest.json
Caching oauth2 token.
Pipeline save succeeded

```

When browse the configuration page of  v2pipetest,  "Automated Triggers" and "Parameters" are empty. 

![image](https://user-images.githubusercontent.com/50065436/58368106-e63ac800-7f1a-11e9-9c7b-f453a4cef0fe.png)

When click "Edit as JSON" then click "RENDERED PIPELINE"

![image](https://user-images.githubusercontent.com/50065436/58368092-a1af2c80-7f1a-11e9-906f-91f9501ac940.png)

Seems inheritance is not working correctly.

### Steps to Reproduce:

See above.

**Comment by louisjimenez** [comment]
Hey @shuai-gao-awx, as of 1.14 pipelines inherit collections such as triggers, notifications, expected artifacts, and parameters by default unless the respective string is included in the `exclude` array.

When a pipeline config is created through spin it does not have the inherited collections saved directly to it. Instead, those collections are lifted from the template when doing a plan. Historically, the UI expects even templated items to be saved to the config and does not look to render what is on the plan. 

V1 relied on items being copied to the config to mitigate this. Although executing the pipeline in v2 should plan correctly, this makes for a confusing UX and i'll be working on some changes for it.

**Comment by shuai-gao-awx** [comment]
Hey @louisjimenez , I got it , thank you for your hard work.

**Comment by louisjimenez** [comment]
The fix for this is now in release 1.14.5. Inherited triggers, notifications, and parameters will render in the UI. To opt out of inheritance, use the `Configure Template` button in the UI or add a matching string,`"exclude": ["parameters", "notifications"]`, to the `exclude` array in the pipeline config.
